### PR TITLE
[luxom] Initial Contribution

### PR DESCRIPTION
**Pull request by jesperskriasoft**
Initial public release of a binding for the https://luxom.io/ IP interface module.
The binding has been tested with the DS65L IP interface, but it’s not an official binding by Luxom.
For the moment only buttons & dimmers are supported.

The communication api is based on the documentation found on the official site :

https://old.luxom.io/uploads/ppfiles/27/LUXOM_ASCII.pdf
https://old.luxom.io/uploads/ppfiles/28/LUXOM_ASCII_extended.pdf
Openhab community thread : https://community.openhab.org/t/new-binding-luxom-io/132343

### CODE REVIEW

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomDimmerHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomDimmerHandler.java:73]: I think `ThingStatusDetail.BRIDGE_OFFLINE` would be more appropriate here, and without the custom text. You cannot configure the thing without a bridge?

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/protocol/LuxomCommunication.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/protocol/LuxomCommunication.java:182]: Should be logger.**warn**.

File: bundles/org.openhab.binding.luxom/src/main/resources/OH-INF/binding/binding.xml
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/resources/OH-INF/binding/binding.xml:8]: I believe the author tag has been dropped.

File: bundles/org.openhab.binding.luxom/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/resources/OH-INF/thing/thing-types.xml:8]: **L**uxom Bridge?

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:289]: As far as I can see this is called from a thread - is it thread-safe?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for this contribution! I have posted a few comments after a first run-through.

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jesperskriasoft) [reply bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:289]: As far as I can see: yes. 

The thread is the loop of the Buffered reader, it will not execute in parallel.

**Comment by jlaur** [comment]
@jesperskriasoft - last commit has a problem with sign-off, seems you used a different e-mail address than previous commits. Also I think you might have missed some of my comments which are hidden by GitHub. :)

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jesperskriasoft) [reply bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:149]: This interval has been given to me by a Luxom developer. So no real need for it, but I'll change it.

**Comment by jesperskriasoft** [comment]
@jlaur the email was a mistake, my intelliJ IDE commit message history didn't have my previous one.  So I copied the format but forgot to change the email address.

**Comment by jesperskriasoft** [comment]
@jlaur I think I covered all your remarks. The DCO plugin still keeps complaining, but in my Intellij commit log all signoffs are correct for now.

**Comment by jlaur** [comment]
> The DCO plugin still keeps complaining, but in my Intellij commit log all signoffs are correct for now

Commit bf5dcbd0c9f3ba4708270af2b643b4cbe4c02030 still has signoff with wrong e-mail address.

File: bundles/org.openhab.binding.luxom/README.md
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/README.md:96]: Please use English names.

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:100]: You can simply use the string "@text/status.connecting":
```suggestion
                    "@text/status.connecting");
```

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:80]: Do not localize logged messages.

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:108]: See other comment - no need to handle localization manually here,

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:115]: See other comment - no need to handle localization manually here,

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:160]: Definitely should not be info, but can probably be omitted entirely. See https://www.openhab.org/docs/developer/guidelines.html#f-logging

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:295]: Consider **trace** level.

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java:305]: `ThingStatus` is enum, so you can compare using `==`.

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomDimmerHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomDimmerHandler.java:76]: I guess no reason to warn about this if configured deliberately because of this issue?

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomDimmerHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomDimmerHandler.java:80]: See other comment - no need to handle localization manually here,

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomSwitchHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomSwitchHandler.java:61]: See other comment - no need to handle localization manually here,

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomThingHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomThingHandler.java:50]: See other comment - no need to handle localization manually here,

File: bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomThingHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomThingHandler.java:108]: See other comment - no need to handle localization manually here,

File: bundles/org.openhab.binding.luxom/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [inline bundles/org.openhab.binding.luxom/src/main/resources/OH-INF/thing/thing-types.xml:37]: First letter uppercase?

**Review by jlaur — CHANGES_REQUESTED** [review]
A few more comments. Sorry that I didn't find much time lately. @openhab/add-ons-maintainers - two review rounds completed, at this point will need to ask for help completing the review of this new binding.

**Comment by jesperskriasoft** [comment]
@jlaur I'm going to make a new merge request because I cannot fix this one for the DCO check...

**Pull request closed without merging**
